WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostListViewModel.kt

```diff
@@ -249,6 +261,37 @@ class ReaderPostListViewModel @Inject constructor(
                 ))
     }
 
+    /**
+     * Handles reblog button action
+     *
+     * @param post post to reblog
+     */
+    fun onReblogButtonClicked(post: ReaderPost) {
+        val selectedSiteId = appPrefsWrapper.getSelectedSite()
+        val selectedSite: SiteModel = siteStore.getSiteByLocalId(selectedSiteId)
```

Comment:
`getSiteByLocalId ` can return a null value (see previous comment in ReblogState); I think we should take care of this possibility.

Reply:
I actually encountered this in the no site use case and handled it in #11553 [here](https://github.com/wordpress-mobile/WordPress-Android/pull/11553/files#diff-4d1136c584e903ce48a0094d9d581b44R269).
I will leave it as is to avoid a conflict for now.

Reply:
Also tackled with the state refactoring 490ae8aab850f4f0e2f9c1e6c38b54d8a80eac07

Reply:
Sharing some thoughts below, I'm not sure if I'm missing something. 

> val selectedSiteId: Int? = appPrefsWrapper.getSelectedSite()

I think selectedSiteId should not be nullable (it can be eventually -1)

> val selectedSite: SiteModel?

I agree selectedSite can be nullable 👍

```
        when (siteStore.visibleSites.size) {
            0 -> _reblogState.value = Event(NoSite)
            1 -> _reblogState.value = selectedSite?.let { Event(PostEditor(it, post)) } ?: Event(Unknown)
            else -> _reblogState.value = selectedSite?.let { Event(SitePicker(it, post)) } ?: Event(Unknown)
        }
```

Maybe we can make it a bit more readable using the when as an expression, something like:

```
       _reblogState.value = when (siteStore.visibleSites.size) {
            0 -> Event(NoSite)
            1 -> if (selectedSite == null) Event(PostEditor(selectedSite, post)) else Event(Unknown)
            else -> if (selectedSite == null) Event(SitePicker(selectedSite, post)) else Event(Unknown)
        }
```

Also `onReblogSiteSelected` I think can have the same issue with `getSiteByLocalId`

wdyt?

Reply:
I agree. Fixed with c9b69578f905b9e8b9bf8e6bd1a8bd1cac57f2d7